aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt

```diff
@@ -0,0 +1,313 @@
+                        pub fn set_express_credentials_provider(&mut self, credentials_provider: #{Option}<#{SharedCredentialsProvider}>) -> &mut Self {
+                            if let #{Some}(credentials_provider) = credentials_provider {
+                                self.runtime_components.set_identity_resolver(#{S3_EXPRESS_SCHEME_ID}, credentials_provider);
+                            }
+                            self
+                        }
+                        """,
+                        *codegenScope,
+                    )
+                }
+
+                else -> emptySection
+            }
+        }
+}
+
+class S3ExpressFluentClientCustomization(
+    codegenContext: ClientCodegenContext,
+) : FluentClientCustomization() {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope =
+        arrayOf(
+            *preludeScope,
+            "S3ExpressRuntimePlugin" to s3ExpressModule(runtimeConfig).resolve("runtime_plugin::S3ExpressRuntimePlugin"),
+        )
+
+    override fun section(section: FluentClientSection): Writable =
+        writable {
+            when (section) {
+                is FluentClientSection.AdditionalBaseClientPlugins -> {
+                    rustTemplate(
+                        """
+                        ${section.plugins} = ${section.plugins}.with_client_plugin(#{S3ExpressRuntimePlugin}::new(${section.config}.config.clone()));
+                        """,
+                        *codegenScope,
+                    )
+                }
+
+                else -> emptySection
+            }
+        }
+}
+
+class S3ExpressRequestChecksumCustomization(
+    codegenContext: ClientCodegenContext,
+    private val operationShape: OperationShape,
+) : OperationCustomization() {
+    private val runtimeConfig = codegenContext.runtimeConfig
+
+    private val codegenScope =
+        arrayOf(
+            *preludeScope,
+            "ChecksumAlgorithm" to RuntimeType.smithyChecksums(runtimeConfig).resolve("ChecksumAlgorithm"),
+            "ConfigBag" to RuntimeType.configBag(runtimeConfig),
+            "DefaultRequestChecksumOverride" to
+                runtimeConfig.awsInlineableHttpRequestChecksum()
+                    .resolve("DefaultRequestChecksumOverride"),
+            "Document" to RuntimeType.smithyTypes(runtimeConfig).resolve("Document"),
+            "for_s3_express" to s3ExpressModule(runtimeConfig).resolve("utils::for_s3_express"),
+        )
+
+    override fun section(section: OperationSection): Writable =
+        writable {
+            when (section) {
+                is OperationSection.AdditionalRuntimePluginConfig -> {
+                    rustTemplate(
+                        """
+                        ${section.newLayerName}.store_put(#{DefaultRequestChecksumOverride}::new(
+                            |original: #{Option}<#{ChecksumAlgorithm}>,
```

Comment:
This should be an inline function so that it doesn't get duplicated for every operation that takes a checksum.

Also, it looks like this is getting added to all operations even if they don't take checksums. This should be limited to the relevant operations.

Reply:
Addressed in 374ed2a